test/box.test.ts

```
@@ -0,0 +1,100 @@
+  test("emoji counts as 2 columns", () => {
+    expect(stringWidth("😀")).toBe(2);
+    expect(stringWidth("hello 😀")).toBe(8); // 5 + 1 + 2
+  });
+
+  test("CJK characters count as 2 columns", () => {
+    expect(stringWidth("你")).toBe(2);
+    expect(stringWidth("hello 你")).toBe(8); // 5 + 1 + 2
+  });
+
+  test("ANSI escape codes have 0 width", () => {
+    const red = "\u001b[31mhello\u001b[39m";
```

ESLint

[error] Use uppercase characters for the value of the escape sequence.

(unicorn/escape-case)